### PR TITLE
sceNp: add error_code and some param checks

### PR DESCRIPTION
**Pull request by Megamouse**
this might be entirely useless, but whatever

### CODE REVIEW

**Comment by digitaldude555** [comment]
> 
> 
> this might be entirely useless, but whatever

Not so fast! 

Urban Trial Freestyle in master gets this in every level.
`U {PPU[0x1000008] Thread (PS3PSNQueueManager) [0x00995c88]} sceNp TODO: sceNpLookupCreateTransactionCtx  `
`U {PPU[0x1000008] Thread (PS3PSNQueueManager) [0x00995ce0]} sceNp TODO: sceNpLookupUserProfileWithAvatarSizeAsync`
`U {PPU[0x1000008] Thread (PS3PSNQueueManager) [0x00995cb0]} sceNp TODO: sceNpLookupPollAsync`
`F {PPU[0x1000000] Thread (main_thread) [0x00145230]} MEM: Access violation reading location 0xfba100a8`

Fixed in your build.
Screenshot (630)

**Comment by elad335** [comment]
Initialization flags need to be either reset on Emu.Stop(); or Emu.Run();
fxm/idm are doing doing resets automatically so..

File: rpcs3/Emu/Cell/Modules/sceNpClans.h [hunk 1]
```diff
@@ -225,8 +225,8 @@ struct SceNpClansMemberEntry
 // Clan message structure
 struct SceNpClansMessage
 {
-	s8 subject[SCE_NP_CLANS_MESSAGE_SUBJECT_MAX_LENGTH + 1];
-	s8 body[SCE_NP_CLANS_MESSAGE_BODY_MAX_LENGTH + 1];
+	char subject[SCE_NP_CLANS_MESSAGE_SUBJECT_MAX_LENGTH + 1];
+	char body[SCE_NP_CLANS_MESSAGE_BODY_MAX_LENGTH + 1];
```
Comment (JohnHolmesII) [inline rpcs3/Emu/Cell/Modules/sceNpClans.h]: Why the change here?

Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/sceNpClans.h]: Can't use strlen on s8.
And it's clear that these are supposed to be chars and this was probably wrong in the first place.

File: rpcs3/Emu/Cell/Modules/sceNp2.cpp [hunk 2]
```diff
@@ -1,511 +1,1390 @@
...
+	}
+
 	return CELL_OK;
 }
 
-s32 sceNpMatching2CreateServerContext()
+error_code sceNpMatching2CreateServerContext(SceNpMatching2ContextId ctxId, vm::cptr<SceNpMatching2CreateServerContextRequest> reqParam, vm::cptr<SceNpMatching2RequestOptParam> optParam, vm::ptr<SceNpMatching2RequestId> assignedReqId)
 {
-	UNIMPLEMENTED_FUNC(sceNp2);
+	sceNp2.todo("sceNpMatching2GetLobbyMemberDataInternal(ctxId=%d, reqParam=*0x%x, optParam=*0x%x, assignedReqId=*0x%x)", ctxId, reqParam, optParam, assignedReqId);
```
Comment (scribam) [inline rpcs3/Emu/Cell/Modules/sceNp2.cpp]:
```suggestion
	sceNp2.todo("sceNpMatching2CreateServerContext(ctxId=%d, reqParam=*0x%x, optParam=*0x%x, assignedReqId=*0x%x)", ctxId, reqParam, optParam, assignedReqId);
```

Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/sceNp2.cpp]: thx
